persistent/Database/Persist/Sql/Types.hs

```diff
@@ -56,6 +59,16 @@ type Migration = WriterT [Text] (WriterT CautiousMigration (ReaderT SqlBackend I
 
 type ConnectionPool = Pool SqlBackend
 
+data ConnectionPoolConfig = ConnectionPoolConfig
+    { connectionPoolConfigStripes :: Int
+    , connectionPoolConfigIdleTimeout :: NominalDiffTime
+    , connectionPoolConfigSize :: Int
+    }
+    deriving (Show)
```

Comment:
I think having a separate data type for the connection pool configuration is nicer than 3 separate function parameters. Thoughts?

Reply:
In an ideal world for me, there is a field of `PostgresConf` named `poolConfig :: ConnectionPoolConfig`. Unfortunately, `PostgresConf` already has an existing field for the pool size, so I broke out stripes + idle timeout to separate fields there.

Reply:
But if adding the two new fields is a breaking change, possibly we could go all-out and replace the fields with a ConnectionPoolConfig field?

Reply:
there are breaking changes, and then there is Breaking Changes, and, weeeeeell, I'd rather the easiest breaking change as possible. Making these types more abstract is a goal of #920 so we can leave a `TODO` here to clean it up with `persistent-3.0` possibly.

Reply:
Ok cool, that's helpful guidance 👍